### PR TITLE
feat: Project selector in EA mode

### PR DESCRIPTION
**Pull request by iamayushm**
# Description

Adding Option to choose project during app creation in EA mode

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Created a project in EA mode and checked if project is assigned or not.
- [x] Give User permission for assigned project and unassigned project and checked under various roles.
- [x] Tested Code for FULL mode.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/charts/modal/DeployChart.tsx [hunk 1]
```diff
@@ -518,7 +518,7 @@ const DeployChart: React.FC<DeployChartProps> = ({
                                     onChange={(e) => setAppName(e.target.value)}
                                 />
                             </label>
-                            {serverMode == SERVER_MODE.FULL && (
+                            {
```
Comment (vivek-devtron) [inline src/components/charts/modal/DeployChart.tsx:521]: We can remove this {

File: src/components/userGroups/UserGroup.tsx [hunk 2]
```diff
@@ -197,8 +197,8 @@ export default function UserGroupRoute() {
         () =>
             Promise.allSettled([
                 getGroupList(),
-                serverMode === SERVER_MODE.EA_ONLY ? null : get(Routes.PROJECT_LIST),
-                serverMode === SERVER_MODE.EA_ONLY ? null : getEnvironmentListMin(),
+                get(Routes.PROJECT_LIST),
+                getEnvironmentListMin(),
```
Comment (vivek-devtron) [inline src/components/userGroups/UserGroup.tsx:201]: Will we get the env list in EA_ONLY?

**Review by vivek-devtron — APPROVED** [review]

File: src/components/v2/values/chartValuesDiff/ChartValuesView.tsx [hunk 3]
```diff
@@ -159,13 +159,13 @@ function ChartValuesView({
...
                     const _fetchedReadMe = commonState.fetchedReadMe
                     _fetchedReadMe.set(0, _releaseInfo.readme)
-
                     dispatch({
                         type: ChartValuesViewActionTypes.multipleOptions,
                         payload: {
                             releaseInfo: _releaseInfo,
                             installedAppInfo: _installedAppInfo,
                             fetchedReadMe: _fetchedReadMe,
+                            selectedProject: _installedAppInfo? _installedAppInfo.teamName ? { value: _installedAppInfo.teamId, label: _installedAppInfo.teamName }  : "" : "",
```
Comment (sohel-rp) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.tsx:168]: We can resolve this to something like - 
```
_installedAppInfo?.teamName ? { value: _installedAppInfo.teamId, label: _installedAppInfo.teamName } : ""
```

File: src/components/userGroups/AppPermissions.tsx [hunk 4]
```diff
@@ -201,7 +201,7 @@ export default function AppPermissions({
                 } as DirectPermissionsRoleFilter
             })
 
-        if (!foundDevtronApps && serverMode !== SERVER_MODE.EA_ONLY) {
+        if (!foundDevtronApps && serverMode != SERVER_MODE.EA_ONLY) {
```
Comment (shivani170) [inline src/components/userGroups/AppPermissions.tsx:204]: Any reason to remove double equals?

File: src/components/v2/values/chartValuesDiff/ChartValuesView.tsx [hunk 5]
```diff
@@ -448,8 +450,12 @@ function ChartValuesView({
                     modifiedValuesYaml: result?.valuesOverrideYaml,
                 },
             })
-        } catch (e) {
+        } catch (e: any) {
             dispatch({ type: ChartValuesViewActionTypes.isLoading, payload: false })
+
+            if (Array.isArray(e.errors)) {
```
Comment (shivani170) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.tsx:456]: Cant we do 
```  if (Array.isArray(e.errors) && e.errors.length > 0) {
            errorMessage = e.errors[0].userMessage
        } else {
            errorMessage = e.message
        }``` intsead

**Comment by iamayushm** [comment]
Duplicate PR. Changes if this PR has been covered in #663.

**Pull request closed without merging**